codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/http/ServerResponseBindingGenerator.kt

```diff
@@ -92,7 +94,7 @@ class ServerResponseBeforeRenderingHeadersHttpBindingCustomization(val codegenCo
                     codegenContext.settings.codegenConfig.publicConstrainedTypes,
                 )
             ) {
-                if (section.context.shape is IntegerShape || section.context.shape is ShortShape) {
+                if (section.context.shape is IntegerShape || section.context.shape is ShortShape || section.context.shape is LongShape || section.context.shape is ByteShape) {
```

Comment:
💡 Maybe we should move this check to a `NumberShape.isIntegral` function?

Reply:
I'd prefer to add an `IntegralShape` class in the hierarchy if we want to dry some of these up further.